### ISSUE TITLE
Not Found trying to add NetNewsWire's own feed to Feedly in NNW.

### ISSUE THREAD

**Issue by StrangeNoises** [Ranchero-Software/NetNewsWire#2443]
This may be a dupe of #2055. Raised as new partly because of the identity of the feed I was having trouble adding, and because it was my very first experience of the app, so potentially an onboarding issue?

NNW 5.1 (3007) for Mac (Big Sur public beta), just downloaded for the first time today because of the new Feedly support, as I've been using Feedly since Google Reader was taken from us, and didn't want to stop (because I also use it on linux/windows).

I noted the "On My Mac" subscriptions on first load, but because I was importing Feedly, I was going to disable that account. But of the subscriptions in it, one I wanted to add to Feedly was of course the one for NetNewsWire itself, to see future blog/announcements about the app.

First thing I tried was to just drag it into the appropriate Feedly folder in the app. "Not found". (That's all the information given in the error dialog.)

Tried copying the feed URL and going to add-feed: "Not found".

Tried copying the home page URL and going to add-feed: "Not found". Tried it pasting that url from the browser url bar instead of as offered by the app. "Not found".

Tried opening Firefox (I've only been trying Safari as default browser during Big Sur public beta), going to the blog homepage site, and clicking on the "Add to Feedly" extension I already have installed on Firefox, added the feed in Feedly. Worked perfectly. Clicked the sync button in the NNW app. It found it and brought it down.

This is the first feed I tried to add to the Feedly account from within NNW rather than its own web-based UI. I'm guessing that's where the problem is (adding feeds to Feedly), as there's presumably nothing about your own feed that your own app wouldn't like. 😉 That also argues against it really being a dupe of #2055, as that looked like being an encoding issue.

**Comment by kielgillard** [comment]
Hello @StrangeNoises, thanks for taking the time to report this. Unfortunately, I can't reproduce it. It could be because I'm not using Big Sur yet.

Are you able to reproduce this issue with another feed? If so, can you please follow these instructions and add the output here?

1. Using the Finder, open the app at this path: Applications > Utilities > Console.app.
1. Using Console.app, choose your computer on “Devices” section of the side bar on the left.
1. From the menu bar, click the “Action” menu.
1. Ensure “Include Info Messages” and “Include Debug” messages are checked.
1. In the search bar, type “Feedly”
1. Use NetNewsWire and attempt to add a feed.

Here's what I see:
<img width="1147" alt="image" src="https://user-images.githubusercontent.com/567949/93845200-50b47180-fce3-11ea-93e8-df40acc617c9.png">

**Comment by StrangeNoises** [comment]
I tried it again just now on my other Mac, which is still running Catalina. Once again, I tried it in turn, dragging it in from the "On My Mac" section, copying the feed url, copying the site url, in no case did it make any difference. Not Found error each time.

This time it was with the Accidentally in Code feed, from the preinstalled defaults, which I also verified was actually from a working site.

This is what I saw in Console, with info/debug enabled as directed:

<img width="1315" alt="Screenshot 2020-09-22 at 10 29 34" src="https://user-images.githubusercontent.com/1295001/93866021-0e793900-fcbf-11ea-957b-8fff74dbd2d7.png">

I see yours is logging the URL it's trying to request, whereas in mine it's just logging "Requesting page for <private>" I'm guessing that's just a redaction in the logging output, either by NNW or by Console? But I don't see anywhere obvious in either to change that.

**Comment by kielgillard** [comment]
Thanks for this, @StrangeNoises. I still need a bit more information, if you could please?

With Console open as per the steps above, can you please manually refresh (File > Refresh or Command + R) your content in NetNewsWire and share with me what you get for the log beginning with "Received collections".

When adding a feed to a folder/collection, I think NetNewsWire might not be identifying the folder/collection correctly, resulting in Feedly saying it is not found.

**Comment by StrangeNoises** [comment]
once again, <private>... (which is why i don't mind just pasting it up here of course...)

<img width="1315" alt="Screenshot 2020-09-22 at 13 52 32" src="https://user-images.githubusercontent.com/1295001/93884558-f7484480-fcda-11ea-8019-52a98804f2be.png">

**Comment by kielgillard** [comment]
Bummer, my console must have been showing me the unredacted logs because I was running a debug build. What's the name of the folder you're dragging or adding the feed to? Does it have any symbols in it, like a `+` or something?

**Comment by StrangeNoises** [comment]
Aha. Yes it does. The destination folder in Feedly is "Tech News + Blogs".

I just tried dragging the same feed as before into a different folder ("News"), and it worked fine. Subsequently trying to drag it from there into the original "Tech News + Blogs" folder fails in the same way though.

**Comment by vincode-io** [comment]
I'm running Big Sur and am unable to recreate this.  I created a "Tech News + Blogs" and blog feed on Feedly and then synced.  I was then able to drag the NetNewsWire item to it.  I think there must be another issue than the "+" in the folder name.

@StrangeNoises Is there anything else unusual about that folder?  Does it have an unusually large number of feeds in it maybe?

**Comment by StrangeNoises** [comment]
There's 19 feeds in that folder *now*; 17 when I started using NNW. It doesn't seem excessive. Well, maybe the name of the feed isn't significant after all, and it's something else. I have no idea.

**Comment by kielgillard** [comment]
I'm planning to do some more checks on my side in the next few days and I'll add some more logging to help debug this issue.

**Comment by kielgillard** [comment]
A quick summary of where this issue is at.

We're waiting on macOS build 5.1.1a2 or later with the improved logging. Getting these unencoded ids in those logs will allow me to encode them against how Feedly encodes them on their website and spot the difference.

From what I gather, Feedly changed the way they identified collections over the last few of years or so. I suspect there is something about older collection ids from Feedly that isn't being encoded properly by NetNewsWire.

**Comment by StrangeNoises** [comment]
OK, with 5.1.1a2, confirm the same problem occurs. Console with the same logging setup as before:

<img width="1365" alt="image" src="https://user-images.githubusercontent.com/1295001/94988026-317ad700-0562-11eb-89e2-3a6dabf85c57.png">

**Comment by kielgillard** [comment]
@StrangeNoises can you please share what you get for the "received collections" log?

> With Console open as per the steps above, can you please manually refresh (File > Refresh or Command + R) your content in NetNewsWire and share with me what you get for the log beginning with "Received collections".

**Comment by StrangeNoises** [comment]
Sorry for the delay, I try to use the machines less in the weekend :-)

```
Received collections: (
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Blogs - Misc",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Entertainments",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Science",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Tech News + Blogs",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Webcomics",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/News",
    "user/723bdc24-ccec-4f1f-b976-8d03de1e8cb0/category/Wine Blogs"
)
```

And the context...

<img width="1460" alt="Screenshot 2020-10-04 at 15 34 37" src="https://user-images.githubusercontent.com/1295001/95018351-1e95fe80-0657-11eb-94f1-88f2d1e2fe7c.png">

At the time I did it there wasn't much to fetch as I'd not long earlier caught up on another machine.

(This is the Big Sur machine.)

**Comment by kielgillard** [comment]
@StrangeNoises Thanks for your patience. While I couldn't reproduce it, I strongly suspect I know what it is going on an a potential fix should be available in the next macOS and iOS releases.

**Comment by kielgillard** [comment]
Nerdy note: I couldn't reproduce it because I can't seem to create a collection/folder on Feedly with a `+` in the collection's identifier; collections these days seem to be created with a UUID as an identifier.

**Comment by StrangeNoises** [comment]
Probably because of problems like this. I started using Feedly when Google Reader was taken down, because they made it easy to transfer my subs at the time. Individual feeds have come and gone but I think most of the folders are the same as they were then! And I probably used + because Google Reader or something wouldn't let me use &. I mean, it's presumably some URL encoding issue along the way, and both those characters have the potential to be problematic in their own ways.

I will deliberately not try renaming it just yet to see if the next release fixes it, for closure's sake. :-)

**Comment by kielgillard** [comment]
@StrangeNoises thanks! I too joined Feedly when the world as we knew it ended so I've got the classic styled collection ids. I tried renaming mine and it retained the previous id. I don't think my encoding changes landed in the 5.1.1 release so let's see how it works in the next release.

**Comment by kielgillard** [comment]
@StrangeNoises can you please check with the latest test build?

**Comment by StrangeNoises** [comment]
It's working for me now. Both via copy/pasting the feed URL into an add-new-feed dialog, and via just dragging it in from the "on my mac" set. (I had no reason to think they'd behave differently, just being sure.) :-)

**Comment by kielgillard** [comment]
@StrangeNoises great news! Thanks so much for your time and patience -- it's made NNW a better app. Much appreciated
